packages/dynamic_layouts/lib/src/dynamic_grid.dart

```diff
@@ -27,15 +26,44 @@ class DynamicGridView extends GridView {
   /// Creates a scrollable, 2D array of widgets that are created on demand.
   DynamicGridView.builder({
     super.key,
+    super.scrollDirection,
+    super.reverse,
     required super.gridDelegate,
     // This creates a SliverChildBuilderDelegate in the super class.
     required super.itemBuilder,
     super.itemCount,
   }) : super.builder();
 
-  // TODO(snat-s): DynamicGridView.wrap?
-
-  // TODO(DavBot09): DynamicGridView.stagger?
+  /// Creates a scrollable, 2D array of widgets where each tile can have a
+  /// varying size in the main axis, generating a staggered layout.
```

Comment:
I think it is important to clarify here that the main extent is based on the child's size, which allows for varying sizes.